R/internals.R

```diff
@@ -400,18 +400,54 @@
 #' loads webpages with a header passed to read_html
 #'
 #' @param page_url url of the page wanted to be loaded
-#'
+#' @param agent_type either "browser" or "desktop"
 #' @return a html webpage
 #'
 #' @noRd
-#'
-.load_page <- function(page_url) {
-  agent <- getOption("worldfootballR.agent", default = "RStudio Desktop (2022.7.1.554); R (4.1.1 x86_64-w64-mingw32 x86_64 mingw32)")
+.load_page <- function(page_url, agent_type = "browser") {
+  agent <- getOption(
+    sprintf("worldfootballR.%s_agent", agent_type),
+    default = "friendly R user coming in peace"
+  )
   ua <- httr::user_agent(agent)
   session <- rvest::session(url = page_url, ua)
   xml2::read_html(session)
 }
 
+
+#' Load transfermarkt page with headers
+#'
+#' loads transfermarkt webpages with a header passed to read_html
+#'
+#' @importFrom httr GET content
+#'
+#' @param page_url url of the page wanted to be loaded
+#' @return a html webpage from transfermarkt
+#'
+#' @importFrom httr GET content add_headers
+#'
+#' @noRd
+.load_page_tm <- function(page_url) {
+  ua_options <- c(
+    'Mozilla/5.0 (X11; Linux x86_64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/51.0.2704.103 Safari/537.36',
+    'Mozilla/5.0 (Windows NT 6.1; Win64; x64; rv:47.0) Gecko/20100101 Firefox/47.0',
+    'Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/91.0.4472.124 Safari/537.36 Edg/91.0.864.59',
+    'Mozilla/5.0 (Macintosh; Intel Mac OS X x.y; rv:42.0) Gecko/20100101 Firefox/42.0',
+    'Mozilla/5.0 (Macintosh; Intel Mac OS X 10_15_7) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/102.0.5005.61 Safari/537.36'
+  )
+
+  ua <- sample(ua_options, 1)
```

Comment:
Not the most intelligent solution to dynamically setting the user agent, but tested on macos locally and all values returned page data...

Thoughts on this @tonyelhabr?

Reply:
It's not bad as a first pass. I found [this package](https://github.com/fangzhou-xie/Randomuseragent) that can randomly generate a valid user agent for us.

In fact, I've just pushed some changes that incorporate this package. The changes also consolidate `.load_page()` and `.load_page_tm()` into just `.load_page()`, always using the randomized browser user agent. The randomized browser user agent should be fine to use for non-transfermarket functions that, up to this point, were using `.load_page()` + the user-specific R desktop agent.

Reply:
It's not bad as a first pass. I found [this package](https://github.com/fangzhou-xie/Randomuseragent) that can randomly generate a valid user agent for us.

In fact, I've just pushed some changes that incorporate this package. The changes also consolidate `.load_page()` and `.load_page_tm()` into just `.load_page()`, always using the randomized browser user agent. The randomized browser user agent should be fine to use for non-transfermarket functions that, up to this point, were using `.load_page()` + the user-specific R desktop agent.